Remove flake-utils

nrabulinski opened the pull request:
Removes flake utils, simplifies the flake definition, and updates the input to latest `nixpkgs-unstable`

nrabulinski committed:
Remove flake-utils

nrabulinski committed:
Change nixpkgs input to nixpkgs-unstable

cdepillabout commented on flake.nix:
```diff
@@ -2,61 +2,39 @@
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ] (system: f nixpkgs.legacyPackages.${system});
+    package =
+      forEachSystem (pkgs:
+        pkgs.haskellPackages.callCabal2nix "example-haskell-nix-flake" ./. {});
+  in {
+    packages = forEachSystem (pkgs: {
+      default =
+        pkgs.haskell.lib.compose.justStaticExecutables
+        package.${pkgs.system};
+    });
+
+    devShells = forEachSystem (pkgs: {
+      default = pkgs.haskellPackages.shellFor {
+        packages = _: [package.${pkgs.system}];
+
+        nativeBuildInputs = [
+          pkgs.cabal-install
+          pkgs.haskellPackages.haskell-language-server
+        ];
+      };
+    });
```
Thanks a lot for fixing this up!

Could you refactor this to make it an overlay (well, a Haskell overlay within a Nixpkgs overlay) on top of Nixpkgs, similar to how the original code is written?

Gerg-L replied:
it doesn't provide an overlay output so that overlay is pointless and requires instantiating a nixpkgs instance

Gerg-L replied:
adding
```nix
    overlays.default = final: prev: {
      OverlayedHaskellPackages = final.haskellPackages.override {
        overrides = hfinal: hprev: {
          # This is our local Haskell package.
          example-haskell-package = hfinal.callCabal2nix "example-haskell-nix-flake" ./. {};
        };
      };
    };
```
will give you a usable external overlay

nrabulinski replied:
I wanted to do that but override the actual Haskell packages and also provide the binary in the overlay, but that's after I come back from vacation so in 3 days

Gerg-L commented:
Use `?ref` and capitalize `NixOS` in the url :trollface:

Gerg-L commented:
Why instantiation a nixpkgs instance for a `packages` output?
IMO have the overlay as it's own output. But don't consume it for your packages

nrabulinski commented:
> Why instantiation a nixpkgs instance for a `packages` output? IMO have the overlay as it's own output. But don't consume it for your packages

because my options are either to repeat myself when declaring the packages or a sketchy abstraction

cdepillabout committed:
Rename haskellPackages to myHaskellPackages

This is to make sure we don't accidentally override Haskell packages
that are used in other places and have very specific transitive
dependencies.  This includes packages like Pandoc and HLS.

cdepillabout committed:
Move devshell into the overlay.

This is so that downstream packages that depend on this flake could
potentially use the development shell.  (Although in that case a better
name is probably waranted...)

cdepillabout committed:
Reformat

cdepillabout committed:
Add workflow for testing flake

cdepillabout commented:
Thanks, I appreciate this!
